data_steward/retraction/retract_utils.py

```diff
@@ -79,6 +79,51 @@ def get_src_id(mapping_type):
     return src_id
 
 
+def get_datasets_list(project_id, dataset_ids):
+    # initialize list of all datasets in project
+    all_dataset_ids = [
+        dataset.dataset_id for dataset in bq.list_datasets(project_id)
+    ]
+
+    if dataset_ids == 'all_datasets':
+        dataset_ids = all_dataset_ids
+    elif dataset_ids == 'none':
+        dataset_ids = []
+    else:
+        dataset_ids = dataset_ids.split()
+        # only consider datasets that exist in the project
+        dataset_ids = [
+            dataset_id for dataset_id in dataset_ids
+            if dataset_id in all_dataset_ids
+        ]
+
+    logging.info('Found datasets to retract from: %s' % ', '.join(dataset_ids))
+    # retract from latest datasets first
+    dataset_ids.sort(reverse=True)
+    return dataset_ids
+
+
+def is_deid_dataset(dataset_id):
+    return bool(re.match(DEID_REGEX, dataset_id)) or bool(
+        re.match(RELEASE_REGEX, dataset_id))
```

Comment:
I just realized there's another function in `retract_utils` that achieves a similar purpose - `get_dataset_type`. Do you think it makes sense to use that instead and remove these functions? I will move all their tests to `get_dataset_type` so that they all pass.

Reply:
I feel like this is something you can do as an additional sub-task to the parent ticket.  It feels a little like scope creep in this PR.  Especially because we can't just get rid of all of these helper functions, (e.g. `get_dataset_type` uses `is_deid_dataset`).  And, I think using the regex pattern is safer than just using the `string1 in string2` check.